python/cugraph-pyg/cugraph_pyg/examples/gcn_dist_sg.py

```diff
@@ -0,0 +1,178 @@
+
+# Must change allocators immediately upon import
+# or else other imports will cause memory to be
+# allocated and prevent changing the allocator
+rmm.reinitialize(devices=[0], pool_allocator=True, managed_memory=True)
+cupy.cuda.set_allocator(rmm_cupy_allocator)
+torch.cuda.memory.change_current_allocator(rmm_torch_allocator)
+
+import torch.nn.functional as F  # noqa: E402
+import torch_geometric  # noqa: E402
+import cugraph_pyg  # noqa: E402
+from cugraph_pyg.loader import NeighborLoader  # noqa: E402
+
+# Enable cudf spilling to save gpu memory
+from cugraph.testing.mg_utils import enable_spilling  # noqa: E402
+
+enable_spilling()
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--hidden_channels", type=int, default=256)
+parser.add_argument("--num_layers", type=int, default=2)
+parser.add_argument("--lr", type=float, default=0.001)
+parser.add_argument("--epochs", type=int, default=4)
+parser.add_argument("--batch_size", type=int, default=1024)
+parser.add_argument("--fan_out", type=int, default=30)
+parser.add_argument("--tempdir_root", type=str, default=None)
+parser.add_argument("--dataset_root", type=str, default="dataset")
+parser.add_argument("--dataset", type=str, default="ogbn-products")
+
+args = parser.parse_args()
+
+wall_clock_start = time.perf_counter()
+device = torch.device("cuda")
+
+from ogb.nodeproppred import PygNodePropPredDataset  # noqa: E402
+
+dataset = PygNodePropPredDataset(name=args.dataset, root=args.dataset_root)
+split_idx = dataset.get_idx_split()
+data = dataset[0]
+
+graph_store = cugraph_pyg.data.GraphStore()
+graph_store[
+    ("node", "rel", "node"), "coo", False, (data.num_nodes, data.num_nodes)
+] = data.edge_index
+
+feature_store = cugraph_pyg.data.TensorDictFeatureStore()
+feature_store["node", "x"] = data.x
+feature_store["node", "y"] = data.y
+
+with tempfile.TemporaryDirectory(dir=args.tempdir_root) as samples_dir:
+    train_dir = os.path.join(samples_dir, "train")
+    os.mkdir(train_dir)
+    train_loader = NeighborLoader(
+        data=(feature_store, graph_store),
+        num_neighbors=[args.fan_out] * args.num_layers,
+        input_nodes=split_idx["train"],
+        replace=False,
+        batch_size=args.batch_size,
+        directory=train_dir,
+    )
+
+    val_dir = os.path.join(samples_dir, "val")
+    os.mkdir(val_dir)
+    val_loader = NeighborLoader(
+        data=(feature_store, graph_store),
+        num_neighbors=[args.fan_out] * args.num_layers,
+        input_nodes=split_idx["valid"],
+        replace=False,
+        batch_size=args.batch_size,
+        directory=val_dir,
+    )
+
+    test_dir = os.path.join(samples_dir, "test")
+    os.mkdir(test_dir)
+    test_loader = NeighborLoader(
+        data=(feature_store, graph_store),
+        num_neighbors=[args.fan_out] * args.num_layers,
+        input_nodes=split_idx["test"],
+        replace=False,
+        batch_size=args.batch_size,
+        directory=test_dir,
+    )
+
```

Comment:
Suggest creating something like `setup_data_loaders ` and  creating the `train/val/test_loader`  by just passing an argument for `input_nodes`

Reply:
fixed